`make generate` failes when not setting GOPATH

In [update-codegen.sh](https://github.com/kubeflow/training-operator/blob/master/hack/update-codegen.sh), the 'GO PATH' is not accessed in a unified way.

For example, `code-generator` and `kube-openapi` was searched under `$(go env GOPATH)`. See:

https://github.com/kubeflow/training-operator/blob/47677112182f60206286c9b72473dade35eb3146/hack/update-codegen.sh#L34

and:

https://github.com/kubeflow/training-operator/blob/47677112182f60206286c9b72473dade35eb3146/hack/update-codegen.sh#L45

`defaulter-gen` was searched under `$GO_PATH`:

https://github.com/kubeflow/training-operator/blob/47677112182f60206286c9b72473dade35eb3146/hack/update-codegen.sh#L88

For developers who haven't set `$GO_PATH` manually, `make generate` will fail:

```console
$ make generate
GOBIN=/mnt/d/workspace/kubeflow/training-operator/bin go install sigs.k8s.io/controller-tools/cmd/controller-gen@v0.10.0
# Is controller gen used for code generation?
/mnt/d/workspace/kubeflow/training-operator/bin/controller-gen object:headerFile="hack/boilerplate/boilerplate.go.txt" paths="./pkg/apis/..."
hack/update-codegen.sh
>> Using /home/zjl/go/pkg/mod/k8s.io/code-generator@v0.25.3
>> Using /home/zjl/go/pkg/mod/k8s.io/kube-openapi@v0.0.0-20220803162953-67bda5d908f1
>> Temporary output directory /tmp/tmp.H8zL9LY8ys
go: downloading k8s.io/gengo v0.0.0-20211129171323-c02415ce4185
Generating deepcopy funcs
Generating clientset for kubeflow.org:v1 at github.com/kubeflow/training-operator/pkg/client/clientset
Generating listers for kubeflow.org:v1 at github.com/kubeflow/training-operator/pkg/client/listers
Generating informers for kubeflow.org:v1 at github.com/kubeflow/training-operator/pkg/client/informers
Generating defaulters for kubeflow.org/v1
hack/update-codegen.sh: line 88: GOPATH: unbound variable
>> Removing /tmp/tmp.H8zL9LY8ys
make: *** [Makefile:45: generate] Error 1
```

--- a/hack/update-codegen.sh
+++ b/hack/update-codegen.sh
@@ -83,9 +83,9 @@ ${CODEGEN_PKG}/generate-groups.sh "all" \
 #echo "Building defaulter-gen"
 #go build -o defaulter-gen ${CODEGEN_PKG}/cmd/defaulter-gen
 
-# ${GOPATH}/bin/defaulter-gen is automatically built from ${CODEGEN_PKG}/generate-groups.sh
+# $(go env GOPATH)/bin/defaulter-gen is automatically built from ${CODEGEN_PKG}/generate-groups.sh
 echo "Generating defaulters for kubeflow.org/v1"
-${GOPATH}/bin/defaulter-gen --input-dirs github.com/kubeflow/training-operator/pkg/apis/kubeflow.org/v1 \
+$(go env GOPATH)/bin/defaulter-gen --input-dirs github.com/kubeflow/training-operator/pkg/apis/kubeflow.org/v1 \
     -O zz_generated.defaults \
     --output-package github.com/kubeflow/training-operator/pkg/apis/kubeflow.org/v1 \
     --go-header-file hack/boilerplate/boilerplate.go.txt "$@" \

